src/index.ts

```diff
@@ -0,0 +1,35 @@
+import { Server } from 'http';
+import app from './app';
+import config from './config/config';
+import db from './config/db';
+import logger from './config/logger';
+
+let server: Server;
+
+const onSigterm = () => {
+  logger.info('SIGTERM received');
+  if (server) {
+    server.close();
+  }
+};
+
+const unexpectedErrorHandler = (error: any) => {
+  logger.error(error);
+  if (server) {
+    server.close(() => {
+      logger.info('Server closed');
+      process.exit(1);
+    });
+  } else {
+    process.exit(1);
+  }
+};
+
+(async () => {
+  await db.connect();
```

Comment:
To lại anh Phương nhở review và xem thử có góp ý gì về DB luôn k nhé người ae
Chỗ này hỏi a Phương xem như nếu viết luôn `await mongoose.connect(config.mongoose.url, config.mongoose.options); `bên cái db.ts kia. Ở đây chỉ gọi lại thôi, k có viết hàm async await như này nữa. Với cách viết như này có gì khác biệt k nhỉ ?

Vì các hàm bên dưới await chỗ này nó xem như là 1 phần callback của hàm connect này rồi

Reply:
nếu viết await thì đâu được, phải async mới await chớ

Reply:
để gọn code thôi chứ ko có khác biệt lắm nhỉ

Reply:
@hungnh-0850 thực ra em viêt luôn mongoose.connect ở đây rồi cho phần còn lại vào callback cũng được. Tuy nhiên hiện tại mình không còn dùng các phiên bản node cũ nữa rồi nên cứ dùng await cho tiện. Hơn nữa sau này application lớn dần ra, sẽ có thể cần viết thêm các function bootstrap trước khi start server nữa, nếu để cả trong callback thì sẽ khá là rối em ạ

Reply:
Dạ, đó là lý do e có cmt này. Tụi e  mong muốn đặt ra những câu hỏi để nhận những cmt từ a để học hỏi thêm về cấu trúc ạ, nên có thời gian nhờ a hỗ trợ giải đáp ạ.
Cám ơn a (bow)